operations/mimir/autoscaling.libsonnet

```diff
@@ -270,36 +276,62 @@
   local cpuHPAQuery(with_ready_trigger) = (
     if with_ready_trigger then
       // To scale out relatively quickly, but scale in slower, we look at the average CPU utilization
-      // per replica over 5m (rolling window) and then we pick the 95th percentile value over the last 15m.
+      // per replica over 5m (rolling window) and then we pick the configured percentile value over the last 15m.
       // We multiply by 1000 to get the result in millicores. This is due to HPA only working with ints.
       //
       // When computing the actual CPU utilization, We only take in account ready pods.
-      |||
-        quantile_over_time(0.95,
-          sum(
-            sum by (pod) (rate(container_cpu_usage_seconds_total{container="%(container)s",namespace="%(namespace)s"%(extra_matchers)s}[5m]))
-            and
-            max by (pod) (min_over_time(kube_pod_status_ready{namespace="%(namespace)s",condition="true"%(extra_matchers)s}[1m])) > 0
-          )[15m:]
-        ) * 1000
-      |||
+      if $._config.autoscaling_cpu_hpa_use_irate then
+        // We use irate() instead of rate() because it is more reliable when scrapes fail. It calculates
+        // the metric from the last two data points in the range, whereas rate() extrapolates missing
+        // values over the full window, which can underestimate CPU utilization and trigger unnecessary
+        // downscaling. To prevent overscaling, we use a lower percentile in irate() compared to rate().
+        |||
+          quantile_over_time(0.65,
+            sum(
+              sum by (pod) (irate(container_cpu_usage_seconds_total{container="%(container)s",namespace="%(namespace)s"%(extra_matchers)s}[5m]))
+              and
+              max by (pod) (min_over_time(kube_pod_status_ready{namespace="%(namespace)s",condition="true"%(extra_matchers)s}[1m])) > 0
+            )[15m:]
+          ) * 1000
+        |||
+      else
+        |||
+          quantile_over_time(0.95,
+            sum(
+              sum by (pod) (rate(container_cpu_usage_seconds_total{container="%(container)s",namespace="%(namespace)s"%(extra_matchers)s}[5m]))
+              and
+              max by (pod) (min_over_time(kube_pod_status_ready{namespace="%(namespace)s",condition="true"%(extra_matchers)s}[1m])) > 0
+            )[15m:]
+          ) * 1000
+        |||
     else
       // To scale out relatively quickly, but scale in slower, we look at the average CPU utilization
```

Comment:
We should update this comment because the way it works is different between rate() and irate(). E.g. in the new query it's **not** the "we look at the average CPU utilization per replica over 5m (rolling window)".